### PR TITLE
Update dependencies in Cargo.lock 

### LINKED ISSUE #1863: Update dependencies in `Cargo.lock`
**Issue summary**

Some of Forest's dependencies have apparently not followed semver and running `cargo update` is causing compilation errors. These errors should be investigated and fixed. Some upper version bounds on dependencies might be required.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1863

**Other information and links**

### CODE REVIEW

**Comment by lemmih** [comment]
Changing the `fvm_shared` dependency to `0.8` solves the root problem?

File: utils/auth/Cargo.toml [hunk 1]
```diff
@@ -6,13 +6,14 @@ edition = "2021"
 
 [dependencies]
 # Public
-fvm_shared   = { version = "0.8", default-features = false }
+fvm_shared   = { version = "=0.8.0", default-features = false }
```
Comment (lemmih) [inline utils/auth/Cargo.toml:9]: Is this still necessary or can we go back to `"0.8"`?

File: utils/auth/Cargo.toml [hunk 2]
```diff
@@ -13,6 +13,7 @@ rand         = "0.8"
 serde        = { version = "1.0", default-features = false, features = ["derive"] }
 thiserror    = "1.0"
 # Internal
+anyhow                = "=1.0.59"
```
Comment (lemmih) [inline utils/auth/Cargo.toml:16]: Is this still required?

Comment (tyshko5) [reply utils/auth/Cargo.toml:16]: Yes. `anyhow` is the reason of our. problems

Comment (lemmih) [reply utils/auth/Cargo.toml:16]: Add the reason why `anyhow` is required as a comment in the `Cargo.toml` file. Right now the dependency looks unused and could easily be deleted.

**Review by lemmih — CHANGES_REQUESTED** [review]

**Review by lemmih — CHANGES_REQUESTED** [review]
This appears to be the root cause: https://github.com/dtolnay/anyhow/issues/230

We can get around this by bumping our toolchain version to `nightly-2022-09-08`. Try this rust version without the restrain on `anyhow`.

**Comment by lemmih** [comment]
Looking good. We do, however, have a minimum dependency on `anyhow = "1.0.64"`. You should add that requirement to `forest/Cargo.toml`.

File: blockchain/chain_sync/src/chain_muxer.rs [hunk 3]
```diff
@@ -165,7 +165,7 @@ where
     M: Provider + Sync + Send + 'static,
     C: Consensus,
 {
-    #[allow(clippy::too_many_arguments)]
+    #[allow(clippy::too_many_arguments, clippy::result_large_err)]
```
Comment (lemmih) [inline blockchain/chain_sync/src/chain_muxer.rs:168]: Would it be too awkward to box the errors?

Comment (tyshko5) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: This doesn't work

Comment (lemmih) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: Please explain why.

Comment (tyshko5) [reply blockchain/chain_sync/src/chain_muxer.rs]: it worked out

Comment (LesnyRumcajs) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: If it worked out then do we need this suppression?

Comment (tyshko5) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: It worked out partly, meaning if we were ok with `map_err`. But David wanted to get of them completely (AFAI understood) and this was not possible (as it propagated all over the codebase). So for now we decided to leave the attribute.

Comment (lemmih) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: I'm 90% sure we can get it to work without having to change too much code. I'll open a new issue for that.

**Review by lemmih — APPROVED** [review]

File: blockchain/chain_sync/src/chain_muxer.rs (hunk 3 above)
Comment (LesnyRumcajs) [reply blockchain/chain_sync/src/chain_muxer.rs:168]: Sounds good

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**